### PR TITLE
Feature/swagger docs

### PR DESCRIPTION
**Pull request by dani6666**

### CODE REVIEW

File: doc/SwaggerApi.xml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+          type: array
+          items:
+            type: object
+            properties:
+              name:
+                type: string
+              profit:
+                description: "Movie income - movie price. May be lower than 0"
+                type: integer
+              ownEnd:
```
Comment (dani6666) [inline doc/SwaggerApi.xml]: Any suggestions for better name than ownEnd?

File: doc/SwaggerApi.xml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+          description: 'Bet successful'
+          schema:
+            $ref: '#/definitions/AuctionDetails'
+        '400':
+          description: Not enough value in bet
+        '403':
+          description: Bad user id
+        '404':
+          description: Auction already ended
+definitions:
```
Comment (sqoshi) [inline doc/SwaggerApi.xml:102]: Why there is no object such a `Video` ?

As long as video in terms of our use  is in some way `abstract` and it consists things like `link`, `title` ,`description`, `views` etc, it should be an object.

Comment (sqoshi) [reply doc/SwaggerApi.xml:102]: Is not that easier to use video like  videoName.link etc ?

File: doc/SwaggerApi.xml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      properties:
+        userId:
+          example: "729b8942-b579-11eb-8529-0242ac130003"
+          type: string
+  UserDetails:
+      type: object
+      properties:
+        cash:
+          type: integer
+        password:
```
Comment (sqoshi) [inline doc/SwaggerApi.xml]: Password?
Aren't we store UUID in some special preferences dir ?

File: doc/SwaggerApi.xml [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+          type: string
+        ownedMovies:
```
Comment (sqoshi) [inline doc/SwaggerApi.xml]: maybe Video?

rentedVideo/ ownedVideo

File: doc/SwaggerApi.xml (hunk 1 above)
Comment (sqoshi) [reply doc/SwaggerApi.xml]: I really do not understandwhat  `ownEnd`  is supposed to be used for, can you maybe add a little bit of description ?

File: doc/SwaggerApi.xml [hunk 5]
```diff
@@ -0,0 +1,165 @@
+swagger: "2.0"
+info:
+  version: 1.0.0
+  title: YnvestTubeApi
+host: "127.0.0.1:8000"
+consumes:
+  - "application/json"
+produces:
+  - "application/json"
+paths:
```
Comment (sqoshi) [inline doc/SwaggerApi.xml:10]: `Paths` are fine, but `endpoints` are just more intuitive.

Comment (sqoshi) [reply doc/SwaggerApi.xml:10]: @dani6666

**Comment by dani6666** [comment]
test comment

File: doc/SwaggerApi.xml (hunk 5 above)
Comment (dani6666) [reply doc/SwaggerApi.xml:10]: This is the swagger language notation. It has to be paths co that swagger can parse it into graphics representation

Comment (sqoshi) [reply doc/SwaggerApi.xml:10]: I thought it is only `illustrative` file my bad, resolved

File: doc/SwaggerApi.xml (hunk 3 above)
Comment (dani6666) [reply doc/SwaggerApi.xml]: True, thats exactly the place where i trolled :D Forgot to remove it from the original version of the document

File: doc/SwaggerApi.xml (hunk 4 above)
Comment (dani6666) [reply doc/SwaggerApi.xml]: Renamed

File: doc/SwaggerApi.xml (hunk 2 above)
Comment (dani6666) [reply doc/SwaggerApi.xml:102]: For the minimal working version we dont really need any endpoint that will return single video or list of them. We haven't got any view to show it yet. We just need to get list of owned videos in getUserDetails endpoint.

In definitions we only have the objects that are only used/returned by any of the endpoints.

File: doc/SwaggerApi.xml [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  /accounts/register:
+    get:
+      summary: "Register new user"
+      operationId: "registerUser"
```
Comment (sqoshi) [inline doc/SwaggerApi.xml]: This is a little personal, but for me it is better to stick to one convention.

I mean you used `accounts` in paths but then in summary and operationId you have used `user`

Maybe change accounts to users or users to accounts.

I feel like it's semantic mistake

File: doc/SwaggerApi.xml (hunk 1 above)
Comment (dani6666) [reply doc/SwaggerApi.xml]: Added the description. This is the date from which user will no longer own this video. We need this to detect how this video should be displayed in the list.

I thought we will show the list of all videos that user ever owned and we will display the videos that he owned but no longer own with grey backgorund of sth like that.

Comment (sqoshi) [reply doc/SwaggerApi.xml]: I would go with videoExpiryDate or , expiryDate

Comment (dani6666) [reply doc/SwaggerApi.xml]: Followed your idea but ended with rentExpirationDate. Should be ok.

File: doc/SwaggerApi.xml (hunk 6 above)
Comment (dani6666) [reply doc/SwaggerApi.xml]: changed paths to users

Comment (dani6666) [reply doc/SwaggerApi.xml]: and removed operationId. I dscovered it is actually senseless and doesn't change anything in the appearance

**Pull request merged**